mtp_noms_ops/apps/prisoner_location_admin/management/commands/load_locations_from_offender_search.py

```diff
@@ -0,0 +1,125 @@
+from mtp_common.nomis import connector, request_retry
+import requests
+
+from prisoner_location_admin.models import PrisonerLocation
+from prisoner_location_admin.tasks import update_locations
+from security.models import PrisonList
+
+logger = logging.getLogger('mtp')
+
+
+class Command(BaseCommand):
+    """
+    Loads prisoner locations from offender search api for all known prisons and submits them to mtp-api
+    """
+    help = __doc__.strip().splitlines()[0]
+
+    def handle(self, **options):
+        try:
+            user = self.get_uploading_user()
+            prison_ids = self.get_known_prison_ids()
+            locations = self.search_for_offenders(prison_ids)
+
+            if locations:
+                self.stdout.write('Scheduling prisoner locations for upload')
+                update_locations(user=user, locations=locations)
+            else:
+                self.stdout.write('Not scheduling prisoner locations for upload')
+        except Exception as e:  # noqa: E722,B001
+            # catch and report any exception to prevent infinite uwsgi spooler loop
+            logger.exception('Prisoner locations not loaded automatically from offender search')
+            if hasattr(e, 'content') and isinstance(e.content, bytes):
+                logger.error(e.content.decode())
+
+    @cached_property
+    def session(self) -> api_client.MoJOAuth2Session:
+        return api_client.get_authenticated_api_session(
+            settings.LOCATION_UPLOADER_USERNAME,
+            settings.LOCATION_UPLOADER_PASSWORD,
+        )
+
+    def get_uploading_user(self) -> MojUser:
+        user_data = self.session.get(f'/users/{settings.LOCATION_UPLOADER_USERNAME}/').json()
+        user = MojUser(user_data['pk'], None, user_data)
+
+        self.stdout.write(f'Starting prisoner location update as {user.username}')
+        return user
+
+    def get_known_prison_ids(self) -> typing.List[str]:
+        prison_list = PrisonList(self.session)
+        prison_ids = [prison['nomis_id'] for prison in prison_list.prisons]
+
+        self.stdout.write(f'{len(prison_ids)} prisons with active prisoner locations')
+        return prison_ids
+
+    def search_for_offenders(self, prison_ids: typing.List[str]) -> typing.List[PrisonerLocation]:
+        locations: typing.List[PrisonerLocation] = []
+
+        headers = connector.build_request_api_headers()
+        for prison_id in prison_ids:
+            self.stdout.write(f'Searching for offenders in {prison_id}…')
+            page = 0
+            while True:
+                url = urljoin(
+                    settings.HMPPS_OFFENDER_SEARCH_BASE_URL,
+                    f'/prison/{prison_id}/prisoners?cellLocationPrefix=&size=500&page={page}&sort=prisonerNumber,ASC',
+                    trailing_slash=False,
+                )
+                response: requests.Response = request_retry(
+                    'get',
+                    url,
+                    retries=0,
```

Comment:
we don't want to retry here? It's a `GET` request without side effects...is it 'cause retrying could have a different set of prisoners and you prefer to rely on the retry mechanism for the whole command?

Reply:
to be honest, _this_ call could totally retry a few times. i just wanted to trial it without any retries anywhere first so that it crashes noisily if there's any problem anywhere